src/main/java/org/jacp/repository/QuestionRepositoryCriteria.java

```diff
@@ -0,0 +1,61 @@
+package org.jacp.repository;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.criteria.*;
+import org.jacp.entity.QuestionEntity;
+import org.jacp.enums.Difficulty;
+import org.jacp.enums.Tags;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author saffchen created on 17.01.2024
+ */
+@Repository
+public class QuestionRepositoryCriteria {
+
+    @PersistenceContext
+    private final EntityManager entityManager;
+
+    public QuestionRepositoryCriteria(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public Optional<List<QuestionEntity>> findByFilter(Pageable pageable, Difficulty difficulty, List<Tags> tags) {
```

Comment:
Do we need Pageable here? How we can use it?

Reply:
To transfer the number of tasks we need to receive

Reply:
Pageable usually used when you need separate something by pages, like 3 page by 10 elements, do you need such functionality here?

Reply:
No, you can do without pageable. Fixed